### PR TITLE
Add CloudVisionDeleteReferenceImageOperator 

### LINKED ISSUE #9558: Missing operator - CloudVisionReferenceImageCreateOperator
Hello,

We miss the operator for method delete_reference_image to hook CloudVisionHook. All other methods probably have matching operators.
This is important to me because the hook method refers to this operator in docstring.
https://github.com/apache/airflow/blob/1d36b0303b8632fce6de78ca4e782ae26ee06fea/airflow/providers/google/cloud/hooks/vision.py#L428

Best regards,
Kamil

### PR DESCRIPTION
**Pull request by manesioz**
Fixes #9558 

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by mik-laj** [comment]
Can you also update docs here?
https://github.com/apache/airflow/blob/1d36b0303b8632fce6de78ca4e782ae26ee06fea/airflow/providers/google/cloud/hooks/vision.py#L428

**Comment by manesioz** [comment]
@mik-laj Yep! Good catch, done.

File: airflow/contrib/operators/gcp_vision_operator.py [hunk 1]
```diff
@@ -206,3 +207,19 @@ def __init__(self, *args, **kwargs):
             DeprecationWarning, stacklevel=3
         )
         super().__init__(*args, **kwargs)
+
+
+class CloudVisionReferenceImageDeleteOperator(CloudVisionDeleteReferenceImageOperator):
```
Comment (mik-laj) [inline airflow/contrib/operators/gcp_vision_operator.py]: We don't need it because this operator was not released in Airflow 1.10

Comment (manesioz) [reply airflow/contrib/operators/gcp_vision_operator.py]: Removed, thanks

**Review by mik-laj — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
@manesioz Can you do a rebase?

**Comment by manesioz** [comment]
@mik-laj Done!

**Comment by mik-laj** [comment]
@menesioz Can you update path in *include directives?
```
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:701: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:706: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:711: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:718: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:726: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:736: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/operators/vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:748: WARNING: Duplicate explicit target name: "howto/operator:cloudvisiondeletereferenceimageoperator".
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:763: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:768: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:773: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:780: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:788: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/example_dags/example_vision.py' not found or reading it failed
/opt/airflow/docs/howto/operator/google/cloud/vision.rst:798: WARNING: Include file '/opt/airflow/docs/airflow/providers/google/cloud/operators/vision.py' not found or reading it failed
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 2]
```diff
@@ -683,6 +683,68 @@ More information
...
+
+For parameter definition, take a look at
+:class:`~airflow.providers.google.cloud.operators.vision.CloudVisionDeleteReferenceImageOperator`
+
+Using the operator
+""""""""""""""""""
+
+We are using the :class:`~google.cloud.vision_v1.types.ReferenceImage` and :class:`~google.api_core.retry.Retry` objects from Google libraries:
+
+.. exampleinclude:: airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
See: https://github.com/apache/airflow/pull/9797

File: docs/howto/operator/google/cloud/vision.rst [hunk 3]
```diff
@@ -680,9 +680,71 @@ Templating
...
+
+For parameter definition, take a look at
+:class:`~airflow.providers.google.cloud.operators.vision.CloudVisionDeleteReferenceImageOperator`
+
+Using the operator
+""""""""""""""""""
+
+We are using the :class:`~google.cloud.vision_v1.types.ReferenceImage` and :class:`~google.api_core.retry.Retry` objects from Google libraries:
+
+.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/providers/google/cloud/example_dags/example_vision.py
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 4]
```diff
@@ -680,9 +680,71 @@ Templating
...
+      :language: python
+      :start-after: [START howto_operator_vision_reference_image_import]
+      :end-before: [END howto_operator_vision_reference_image_import]
+
+.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/providers/google/cloud/example_dags/example_vision.py
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 5]
```diff
@@ -680,9 +680,71 @@ Templating
...
+      :language: python
+      :start-after: [START howto_operator_vision_retry_import]
+      :end-before: [END howto_operator_vision_retry_import]
+
+.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/providers/google/cloud/example_dags/example_vision.py
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 6]
```diff
@@ -680,9 +680,71 @@ Templating
...
+      :language: python
+      :start-after: [START howto_operator_vision_reference_image]
+      :end-before: [END howto_operator_vision_reference_image]
+
+The ``product_set_id`` argument can be omitted (it will be generated by the API):
+
+.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/providers/google/cloud/example_dags/example_vision.py
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 7]
```diff
@@ -680,9 +680,71 @@ Templating
...
+      :language: python
+      :dedent: 4
+      :start-after: [START howto_operator_vision_reference_image_delete]
+      :end-before: [END howto_operator_vision_reference_image_delete]
+
+Or it can be specified explicitly:
+
+.. exampleinclude:: /../airflow/airflow/providers/google/cloud/example_dags/example_vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. exampleinclude:: /../airflow/providers/google/cloud/example_dags/example_vision.py
```

File: docs/howto/operator/google/cloud/vision.rst [hunk 8]
```diff
@@ -680,9 +680,71 @@ Templating
...
+      :language: python
+      :dedent: 4
+      :start-after: [START howto_operator_vision_reference_image_delete_2]
+      :end-before: [END howto_operator_vision_reference_image_delete_2]
+
+
+Templating
+""""""""""
+
+.. literalinclude:: ../../../../airflow/providers/google/cloud/operators/vision.py
```
Comment (mik-laj) [inline docs/howto/operator/google/cloud/vision.rst]:
```suggestion
.. literalinclude:: /../airflow/providers/google/cloud/operators/vision.py
```

**Comment by mik-laj** [comment]
I accepted my suggestions because it is an obvious typo and I would like to check it on CI. It does not affect the behavior or the final result.

**Comment by manesioz** [comment]
Thank you @mik-laj ! Sorry for the typo. It's ready to merge whenever you are :)

**Comment by mik-laj** [comment]
@manesioz What is your plan for the next contribution?

**Comment by manesioz** [comment]
I'll continue to look for manageable tickets -- I'd love to continue contributing as much as possible. If you have any ideas feel free to reach out :)

**Pull request merged**